### PR TITLE
feat: support font-face @import

### PR DESCRIPTION
**Pull request by blaggacao**
closes: #16

### CODE REVIEW

**Comment by blaggacao** [comment]
@paulzi This eventually fetches the right text:
image

I'm no JS expert, can you help me to get the async logic ironed out? This would then close #16

File: providers/font-face/FontFaceSessionProvider.js [hunk 1]
```diff
@@ -53,6 +64,15 @@ export default class FontFaceSessionProvider extends FontsSessionProvider {
     #parseStyleSheet(styleSheet) {
         for (let i = 0, len = styleSheet.cssRules.length; i < len; i++) {
             let rule = styleSheet.cssRules[i];
+            if (rule.constructor.name === 'CSSImportRule') {
+                let doc = document.implementation.createHTMLDocument(""),
+                  styleElement = document.createElement("style");
+                this.#loadData(rule.href).then((css) => {
+                    styleElement.textContent = css;
+                });
```
Comment (blaggacao) [inline providers/font-face/FontFaceSessionProvider.js:72]: How can we wait here for the promise to resolve, first?

Comment (blaggacao) [reply providers/font-face/FontFaceSessionProvider.js:74]: Looks like this did the trick: https://github.com/paulzi/svg-text-to-path/compare/96f2f2db776e33a15584c477ae299e94e0f88ff1..1a56e0b9402e5a87b127187bd564fd45c8fce472

**Comment by paulzi** [comment]
I have verified that the browser security CORS can be override by simply adding attribute `crossorigin="anonymous"` to `<link>` tag.
Technically, this can be done by writing the attribute forcibly:
```javascript
for (let i = 0; i < document.styleSheets.length; i++) {
    let stylesheet = document.styleSheets[i];
    let clone = stylesheet.ownerNode.cloneNode();
    clone.setAttribute('crossorigin', 'anonymous');
    stylesheet.ownerNode.outerHTML = clone.outerHTML; 
}
```

However, it seems to me that it is not entirely correct to do this for the user.

Your solution with ajax loading of styles is interesting, special thanks for `DOMImplementation.createHTMLDocument()`, I did not know about this function. But this will again cause the styles to be reloaded, as well as all nested imports.

I think the best solution is here:
1) Give information in the readme that if you load styles from external sources, without configuring cors, then add attribute `crossorigin`.
2) Add an optional parameter for crossorigin override.
3) Add try-catch when reading the rules

**Comment by paulzi** [comment]
I need a couple of days to implement a flexible and correct solution, considering asynchronous loading.

**Comment by paulzi** [comment]
Closed and rewrited in v2.0.2.

**Pull request closed without merging**